# Os conceitos de **Média**, **Mediana** e **Moda**

Média, mediana e moda são **medidas de posição** que visam **representar os fenômenos por seus valores centrais, em torno dos quais tendem a concentrar-se os dados**. Antes de calcular essas medidas, é importante considerar como os dados podem se aprenentar:

- **Agrupados**: Quando estão dispostos em uma distribuição de frequência;
- **Não Agrupados**: Quando estão dispostos em rol ou dados brutos.

# Média ($\bar{X}$)

## Para dados não agrupados

$\bar{X} = \frac{\sum_{i=1}^{n} X_i}{n}$

## Para dados agrupados

Quandos os dados se apresentam em distribuição de frequência, a média é definida por:

$\bar{X} = \frac{\sum{i=1}^{n} P_iF_i}{n}$

Onde:

- $P_i$ é o ponto médio da classe i;
- $F_i$ é a frequência absoluta da classe i;
- n é o tamanho do conjunto de dados ou da amostra.

Exemplo:

Considere os dados sobre peso(em kg) de recém-nascidos de certa maternidade, dispostos
na seguinte distribuição de frequência:

| Classe (Notas) | Frequência Absoluta |
| -------------- | ------------------- |
| [2.0, 2.5[     | 2                   |
| [2.5, 3.0[     | 4                   |
| [3.0, 3.5[     | 7                   |
| [3.5, 4.0[     | 5                   |
| [4.0, 4.5[     | 5                   |
| [4.5, 5.0]     | 7                   |
| Soma           | 30                  |



In [1]:
tabela = {
    (2.0, 2.5): 2,
    (2.5, 3.0): 4,
    (3.0, 3.5): 7,
    (3.5, 4.0): 5,
    (4.0, 4.5): 5,
    (4.5, 5.): 7
}

ponto_medio = lambda limite_superior, limite_inferior: (limite_superior + limite_inferior)/2

PiFi = [
    ponto_medio(limite_superior, limite_inferior) * frequencia_absoluta
    for (limite_inferior, limite_superior), frequencia_absoluta in tabela.items()
]

media_agrupada = round(sum(PiFi) / sum(tabela.values()), 2)

print(media_agrupada)

3.72


# Mediana ($M_d$)

## Para dados não agrupados

Em um rol, a mediana é o elemento que ocupa a posição central, de forma que metade dos
dados está acima e a outra metada está abaixo da mediana.

Para o cálculo da mediana para dados não agrupados, são levados em consideração dois
fatores:

1. Se o tamanho da amostra é ímpar ou par;
2. Se tem relaão com **elemento mediano**, que é o elemento que nos dá a posição ocupada pela mediana.

### Se n é ímpar

A mediana será o valor dado pela **posição** determinada pelo
**elemento mediano ($E_{M_{d}}$)**:

$E_{M_{d}}$ = $\frac{n + 1}{2}$

Exemplo: considere o seguinte conjunto de dados [2,5,7,9,10]. Assim, como n=5, temos:

$E_{M_{d}}$ = $\frac{5 + 1}{2}$ = 3 $\rightarrow$ $M_d$ = 7

### Se n é par

A mediana será a **média aritmética entre as medianas obtidas pela posição dos 
elementos medianos $E_{M_{d_{1}}}$ e $E_{M_{d_{2}}}$**, que são derminados da seguinte
forma (note que nesse caso n também o número de elementos do conjunto):

$
\begin{cases}
E_{M_{d_{1}}} = \frac{n}{2} \rightarrow M_{d_{1}} \\
E_{M_{d_{2}}} = \frac{n}{2} + 1 \rightarrow M_{d_{2}}
\end{cases}
\quad \Rightarrow \quad
M_d = \frac{M_{d_{1}} + M_{d_{2}}}{2}
$

Exemplo: 3,4,6,8,10,11

$
\begin{cases}
E_{M_{d_{1}}} = \frac{6}{2} = 3 \rightarrow M_{d_{1}} = 6 \\
E_{M_{d_{2}}} = \frac{6}{2} + 1 = 4 \rightarrow M_{d_{2}} = 8
\end{cases}
\quad \Rightarrow \quad
M_d = \frac{M_{d_{1}} + M_{d_{2}}}{2} = \frac{6 + 8}{2} = 7
$

## Para dados Agrupados

1. Determinar o elemento mediano (aqui não importa a paridade):

    $E_{M_{d}} = \frac{n}{2}$
    
2. Determinar a classe mediana ($C_{M_{d}}$), que é **a classe que contém o elemento
    mediano**. Note que para isso precisamos calcular também a frequência acumulada.
    
3. Aplicar a fórmula:

   $M_{d} = L_{i_{M_{d}}} + (\frac{E_{M_{d}} - F_{aac}}{F_{M_{d}}}) * h$
   
Onde: 

- $L_{i_{M_{d}}}$ = Limite inferior da classe mediana;
- $F_{acc}$ = Frequência acumulada **anterior** à classe mediana;
- $F_{M_{d}}$ = Frequência absoluta da classe mediana;
- h = Amplitude da classe mediana (diferença entre os limites da classe mediana) 


Exemplo: Considerando os dados dos pesos dos recém-nascidos na tabela adiante, vamos
determinar o valor da mediana.

| Classe (Notas) | Frequência Absoluta | Frequência Acumulada |
| -------------- | ------------------- | -------------------- |
| [2.0, 2.5[     | 2                   | 2                    |
| [2.5, 3.0[     | 4                   | 6                    |
| [3.0, 3.5[     | 7                   | 13                   |
| [3.5, 4.0[     | 5                   | 18                   |
| [4.0, 4.5[     | 5                   | 23                   |
| [4.5, 5.0]     | 7                   | 30                   |
| Soma           | 30                  | -                    |



1. $E_{Md} = \frac{30}{2} = 15 \rightarrow$ **Classe Mediana = [3.5, 4.0[** pois é a
    classe que contém do décimo quarto até o décimo oitavo elementos
    (é fácil verificar que o décimo quinto se encontra nesse limite);
3. $L_{i_{M_{d}}} = 3.5$;
4. $F_{acc} = 13$;
5. $F_{M_{d}} = 5$;
6. h = 4 - 3.5 = 0.5;
7. $M_{d} = 3.5 + (\frac{15 - 13}{5}) * 0.5 = 3.7$ 


# Moda ($M_o$)

Moda é o valor mais frequente no conjunto de dados. Caso nenhum valor se repita,
dizemos que o conjunto de dados é amodal.

## Para dados não agrupados

Exemplo 1:

1,2,2,3,5 &rarr; $M_o$ = 2, veja que o número 2 é o valor que mais se repete.

Exemplo 2:

1,3,6,7 &rarr; Amodal, pois não temos nenhuma repetição dos valores.

Exemplo 3:

1,2,3,3,4,5,5 &rarr; $M_o$ = 3 e 5 (bimodal).

## Para dados agrupados

1. Determinar a **classe modal $C_{M_{o}}$**, que é a classe com maior frequência,
    seja ela absoluta ou relativa.
2. Calcular a moda a partir da fórmula de Czuber:

$M_o = L_{I_{M_{o}}} + \frac{\Delta_{1}}{\Delta_{1} + \Delta_{2}} * h$

Onde:

- $L_{I_{M_{o}}}$ = Limite inferior da classe modal;
- $\Delta_{1}$ = frequência absoluta da classe modal - frequência absoluta da classe 
imediatamente inferior;
- $\Delta_{2}$ = frequência absoluta da classe modal - frequência absoluta da classe 
imediatamente posterior;
h = Amplitude da classe modal.

Exemplo: Considerando novamente os dados dos pesos dos recém nascidos, obtenha o valor
da moda.

| Classe (Notas) | Frequência Absoluta | 
| -------------- | ------------------- | 
| [2.0, 2.5[     | 2                   | 
| [2.5, 3.0[     | 4                   | 
| [3.0, 3.5[     | 7                   | 
| [3.5, 4.0[     | 5                   | 
| [4.0, 4.5[     | 5                   | 
| [4.5, 5.0]     | 7                   | 
| Soma           | 30                  | 


&rarr; $C_{M_{o}}$ = [3.0, 3.5[ e [4.5, 5.0[. O conjunto é bimodal!

### Calculando primeira moda [3.0, 3.5[
 
2. $L_{I_{M_{o}}} = 3.0$;
3. $\Delta_{1} = 7 - 4 = 3$;
4. $\Delta_{2} = 7 - 5 = 2$;
5. h = 0.5
6. Fórmula de Czuber: $3 + (\frac{3}{3 + 2}) * 0.5 = 3.3$

### Calculando segunda moda [4.5, 5.0[

2. $L_{I_{M_{o}}} = 4.5$;
3. $\Delta_{1} = 7 - 5 = 2$;
4. $\Delta_{2} = 7 - 0 = 7$;
5. h = 0.5
6. Fórmula de Czuber: $4.5 + (\frac{2}{2 + 7}) * 0.5 = 4.61$
